codalab/worker_manager/worker_manager.py

```diff
@@ -143,7 +143,8 @@ def run_one_iteration(self):
 
             # Make sure we don't queue up more workers than staged UUIDs if there are
             # more workers still booting up than staged bundles
-            if len(pending_worker_jobs) >= len(self.staged_uuids):
+            # However override this if we have less than minimum number of workers
+            if len(pending_worker_jobs) >= len(self.staged_uuids) and not less_than_min_workers:
```

Comment:
Is there a way to make this logic a bit easier to follow, since `less_than_min_workers` showing up again here makes things kind of coupled?

Reply:
Ok changed the logic a little bit.